cwl_tools/basicfiltering/filter_vardict.py

```diff
@@ -112,11 +114,30 @@ def run_std_filter(args):
     nsampleName = vcf_reader.samples[1]
 
     vcf_writer = vcf.Writer(open(vcf_out, 'w'), vcf_reader)
+    vcf_complex_writer = vcf.Writer(open(vcf_complex_out, 'w'), vcf_reader)
     txt_fh = open(txt_out, "wb")
     # Iterate through rows and filter mutations
     for record in vcf_reader:
         tcall = record.genotype(args.tsampleName)
 
+        # Pad complex indels for proper genotyping
+        if (record.INFO["TYPE"] == "Complex" and 
+                len(record.REF) != len(record.ALT) and 
+                record.INFO["SHIFT3"] > 0 and 
+                record.INFO["SHIFT3"] <= len(record.INFO["LSEQ"])):
+            padding_seq = record.INFO["LSEQ"][len(record.INFO["LSEQ"])-(record.INFO["SHIFT3"]+1):]
+            record.REF = padding_seq + record.REF
+            for alt in record.ALT:
+                alt.sequence = padding_seq + alt.sequence
+            record.POS = record.POS - (record.INFO["SHIFT3"]+1)
+            # TODO: add ID=SHIFT3_ADJUSTED to vcf header
+            record.INFO["SHIFT3_ADJUSTED"] = record.INFO["SHIFT3"]
+            record.INFO["SHIFT3"] = 0
+            complex_flag = True
+        else:
+            complex_flag = False
+            record.INFO["SHIFT3_ADJUSTED"] = 0
```

Comment:
I've tested this, and noticed that the `SHIFT3_ADJUSTED` field does not always come out as `0` in the non-complex case

I'm not sure why, perhaps the record object's info is not actually being updated here?